lib/std/builtin.zig

```diff
@@ -664,16 +664,17 @@ pub const PanicFn = fn ([]const u8, ?*StackTrace) noreturn;
 
 /// This function is used by the Zig language code generation and
 /// therefore must be kept in sync with the compiler implementation.
-pub const panic: PanicFn = if (@hasDecl(root, "panic")) root.panic else default_panic;
+pub const panic: PanicFn = if (@hasDecl(root, "panic"))
+    root.panic
+else if (@hasDecl(root, "os") and @hasDecl(root.os, "panic"))
+    root.os.panic
+else
+    default_panic;
 
 /// This function is used by the Zig language code generation and
 /// therefore must be kept in sync with the compiler implementation.
 pub fn default_panic(msg: []const u8, error_return_trace: ?*StackTrace) noreturn {
     @setCold(true);
-    if (@hasDecl(root, "os") and @hasDecl(root.os, "panic")) {
-        root.os.panic(msg, error_return_trace);
```

Comment:
Why did you remove the error return trace here?

Reply:
This code isn't needed because default_panic is no longer used when `root.os.panic` is defined.

Reply:
Currently the language reference does not mention os-specific panic handler or what the expected behavior is, so I feel unsure about this. 
I would suggest changing doc.langref.html.in starting at `{#header_open|@panic#}` to force clarification on this or ping/talk to somebody who knows how it stuff was agreed to work and quote it.

Reply:
Yeah unfortunately all of this is undocumented. In theory I think the docs would go here: https://ziglang.org/documentation/master/#Root-Source-File